core_cmds.c

```diff
@@ -967,9 +969,9 @@ static int w_force_tcp_alias(struct sip_msg *msg, int *port)
 	return 1;	
 }
 
-static int w_set_adv_address(struct sip_msg *msg, str *adv_addr)
+static int w_set_adv_address(struct sip_msg *msg, str *adv_addr, str *adv_addr_via)
```

Comment:
I'm presuming with this change we can still use the function as before and this isn't a breaking change?

Reply:
Correct. The `adv_addr_via` parameter is optional ([see above](https://github.com/purecloudlabs/opensips/pull/80/files#diff-334e35a876eac1b72d4bbec45904ca4b4e284bfacd805af212bb2d4a51d081a2R223)). Therefore, its length should be 0 if it is not provided when calling `set_advertised_address(...)`.

Reply:
But pointing this out, I just revisited the above code, and I think it is safer to initialize `msg->set_global_address_via.len` in this function as well: 84b6605.

Reply:
Will test with our stack for regression before merge.

Reply:
Actually this was a good catch, as we had a regression: d1bd03dfd319cdb2195367043605fa1651daad71